handler/request.py

```diff
@@ -0,0 +1,112 @@
+from flask import jsonify
+from dao.request import Request
+from handler.user import BaseHandler
+
+
+class RequestHandler(BaseHandler):
+    # def __init__(self):
+    #     print("lol")
+
+    # def build_request_dict(self, row):  # row is a python list
+    #     result = {}
+    #     result['rid'] = row[0]
+    #     result['supplyName'] = row[1]
+    #     result['time'] = row[2]
+    #     result['status'] = row[3]
+    #     result['description'] = row[4]
+    #     result['uid'] = row[5]
+    #     result['did'] = row[6]
+    #     return result
+
+    @staticmethod
+    def get_all_requests():
+        try:
+            requests = Request().get_all_requests()
+            result_list = []
+            for row in requests:
+                result_list.append(row.to_dict())
+            result = {
+                "message": "Success!",
+                "requests": result_list,
+            }
+            print(result_list)
+            return jsonify(result), 200
+        except:
+            return jsonify(message="Server error!"), 500
+
+    @staticmethod
+    def get_request_by_id(rid):
+        if rid > 0:
```

Comment:
Comparison with 0 is because there should never be a negative number rid. It will work but the front end group would have to be wary... and also when testing from postman (aka the professor)